### PR TITLE
[nwprov] Make it possible to notify network commissioning cluster for autonomous connection

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
LastNetworkID, LastNetworkingStatus, LastConnectErrorValue should be handled by driver for returning "the last attempt to connect to an operational network, using this interface, whether by invocation of the ConnectNetwork command or by autonomous connection after loss of connectivity or during initial establishment."

#### Change overview
- Update DriverAPI
- Update Linux Driver and GenericThreadDriver

#### Testing
- Updated test

### CODE REVIEW

File: src/include/platform/NetworkCommissioning.h [hunk 1]
```diff
@@ -175,6 +175,25 @@ class BaseDriver
      */
     virtual NetworkIterator * GetNetworks() = 0;
 
+    /**
+     * @brief Returns the status of last network operations (i.e. scan and attach attempts.) If no such attempts are conducted,
+     * CHIP_ERROR_KEY_NOT_FOUND will be returned.
+     */
+    virtual CHIP_ERROR GetLastNetworkingStatus(Status & status) { return CHIP_ERROR_KEY_NOT_FOUND; }
```
Comment (bzbarsky-apple) [inline src/include/platform/NetworkCommissioning.h:182]: Don't these need to be implemented for all the drivers, not just Linux + Thread, if we're going to remove the shared cross-platform impl?

Comment (erjiaqing) [reply src/include/platform/NetworkCommissioning.h:182]: Yes, other platforms should also implement this.

File: src/platform/P6/NetworkCommissioningDriver.h [hunk 2]
```diff
@@ -94,6 +94,11 @@ class P6WiFiDriver final : public WiFiDriver
     CHIP_ERROR Init() override;
     CHIP_ERROR Shutdown() override;
 
+    // TODO: Implement this
```
Comment (andy31415) [inline src/platform/P6/NetworkCommissioningDriver.h:97]: please add a github issue number for these TODOs, either one central one with checkboxes or one per platform

Comment (erjiaqing) [reply src/platform/P6/NetworkCommissioningDriver.h]: #15700

**Review by tcarmelveilleux — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**